File: LeetCode-September-Challenge/N-aryTreeLevelOrderTraversal.cpp
#include<bits/stdc++.h>
using namespace std;
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        if(!root) return {};
        queue<Node *> q;
        q.push(root); vector<vector<int>> res;
        while(!q.empty()){
            int size = q.size();
            vector<int> temp;
            while(size--){
                Node *curr = q.front();
                q.pop();
                for(auto child : curr->children) q.push(child);
                temp.push_back(curr->val);
            }
            res.push_back(temp);
        }
        return res;
    }
};

File: LeetCode-September-Challenge/NaryTreeLevelOrderTraversal.js
function Node(val, children) {
  this.val = val;
  this.children = children;
}

var levelOrder = function (root) {
    if(!root) return []; let q = [];
    q.push(root); let res = [];
    while(q.length > 0){
        let size = q.length;
        let temp = new Array();
        while(size--){
            let curr = q.shift();
            for(let i = 0; i < curr.children.length; ++i) q.push(curr.children[i]);
            temp.push(curr.val);
        }
        res.push(temp);
    }
    return res;
};
var levelOrder = function (root) {
    if(!root) return []; let q = [root]; let res = [];
    while(q.length > 0){
        let size = q.length;
        let temp = new Array();
        while(size--){
            let curr = q.shift();
            q.push(...curr.children);
            temp.push(curr.val);
        }
        res.push(temp);
    }
    return res;
};

